tools/telemetry-generator/test/appInsightsMemoryUsageTestHandler.test.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TelemetryClient } from "applicationinsights";
+const handler = require("../src/handlers/appInsightsMemoryUsageTestHandler");
+
+describe("appInsightsMemoryUsageTestHandler", () => {
+	let mockTelemetryClient: TelemetryClient;
+	let mockFileData: any;
+
+	beforeEach(() => {
+		mockTelemetryClient = {
+			trackMetric: jest.fn(),
+		} as unknown as TelemetryClient;
+
+		mockFileData = {
+			suiteName: "TestSuite",
+			benchmarks: [
+				{
+					benchmarkName: "Benchmark1",
+					customData: {
+						"Heap Used Avg": "123.45",
+						"Heap Used StdDev": "67.89",
+					},
```

Comment:
We should do this for all tests. The JSON files we expect have plain numbers, not strings, so we should probably test like that too.
```suggestion
					customData: {
						"Heap Used Avg": 123.45,
						"Heap Used StdDev": 67.89,
					},
```